backend/src/entities/product.ts

```diff
@@ -1,33 +1,32 @@
-import { Types } from "mongoose";
+// src/entities/product.ts
+
+import { Document, Types } from "mongoose";
 import { User } from "./user.js";
 
 export enum ProductType {
     CROP = 1,
     POULTRY = 2,
 };
 
-export type ProductDAO = {
-    _id: Types.UUID;
+export interface ProductDAO extends Document {
```

Comment:
Why extend `Document`? The DAO is for the data in the document, not the document itself. Same with `TransactionDAO`?